mathesar_ui/src/header/Header.svelte

```diff
@@ -15,17 +18,27 @@
   import {
     Icon,
     Button,
+    Dropdown,
   } from '@mathesar-components';
 
   import SchemaSelector from './schema-selector/SchemaSelector.svelte';
   import ImportIndicator from './import-indicator/ImportIndicator.svelte';
 
-  function createNewTable() {
+  async function createEmptyTable() {
+    const response = await postAPI<{ id: number, name: string }>('/tables/', {
+      schema: $currentSchemaId as number,
+    });
+    const { id, name } = response;
+    await refetchSchema($currentDBName, $currentSchemaId);
```

Comment:
This logic can be moved to `tables` store (/mathesar_ui/src/stores/tables.ts).

`refetchSchema` currently only fetches schema related information, and not the tables within the schema. The `refetchTablesForSchema` method within `tables` store can be used to refetch the table list.

As for the type definition for the post response, the type `TableEntry` can be used.